Feature/user update

VincentProuchet opened the pull request:
La feature demandée est complète et fonctionne comme prévue

Il me faut créer le test unitaire
je vais en profiter pour apprendre les test unitaire en node.js

j'ai résolu les conflit de merge
principalement en prenant tout de incoming

vérifie quand même 
j'ai eu un couac dans user.controler avec des methodes en double.

VincentProuchet committed:
Creation/implementatiobn update => user.controler

creation et implémentation methode update => user.service
refactoring  User.entity
    ajout proprièté pertinentes

Mise à jour schéma Database.drawio
Mise à jour schéma API_endpoint.drawio

VincentProuchet committed:
Corections

mise à jour
    User.entity=> décoration Exclude()
    UserUpdateDTO=> il fallait des décorateur
        pour que le controleur puisse faire les conversions
    user.mapper.dto=> ajour d'une fonction de conversion
        entity=> UserUpdateDto

Nettoyage des imports
yarn lint

les test avec postman montre que tout fonctionne comme attendu

VincentProuchet committed:
Merge branch 'dev' of github.com:Guuzii/Nest-api-template into feature/user-update

VincentProuchet committed:
retrait de doublons dans user.controler

mais il y a des problèmes dans les test unitaire
et un probléme avec les fichiers de migrations
je sais pas d'ou vien l'erreur
    can't import on non module file

Guuzii commented:
J'aurais besoin que tu fasse un lint et format parce que la tu modifie tous les fichiers avec tes indentations de 4 espaces

VincentProuchet committed:
yarn format

yarn lint
yarn lint indique 3 erreurs
plusieur variables non utilisée=>user.service.spec

VincentProuchet committed:
Merge branch 'dev' of github.com:Guuzii/Nest-api-template into feature/user-update

re yarn lint
re yarn format

VincentProuchet commented:
done

indent : tab
size  2

je viens de terminer
j'ai changé la régle de no-unused-var danqs eslintrc.ts
on la remettras vers la fin pour nettoyer et optimiser le code.

VincentProuchet committed:
yarn lint/format

modifié .eslintrc.js
  retiré les variables non utilisées des contrôles
    sinon on vas passer notre temps à kes retirer pour les ré-ajouter
    l'optimiasation des variables non-utilisées
    c'est le dernier stade de l'optimisation
